src/account/UpgradeableModularAccount.sol

```diff
@@ -482,10 +526,28 @@ contract UpgradeableModularAccount is
         // exists.
         for (uint256 i = 0; i < hooksLength; ++i) {
             bytes32 key = executionHooks.at(i);
-            (FunctionReference hookFunction, bool isPreHook, bool isPostHook,) = toExecutionHook(key);
+            (FunctionReference hookFunction, bool isPreHook, bool isPostHook, bool requireUOContext) =
+                toExecutionHook(key);
+
+            if (!isPackedUO && requireUOContext) {
+                revert RequireUserOperationContext();
```

Comment:
For normal exec hooks need UO context, would it run not in this flow and revert the call?

Reply:
Are you saying it would revert here, because we are currently calling the exec hooks twice (once with UO context as part of executeUserOp() and once via subsequent self-call within fallback)?

Reply:
Discussed offline!